src/main/java/com/google/sps/servlets/utils/CommentAnalysis.java

```diff
@@ -53,31 +53,56 @@ public CommentAnalysis(LanguageServiceClient languageService) {
    * @return a Statistics object that contains required values to display
    */
   public Statistics computeOverallStats(CommentThreadListResponse youtubeResponse) {
-    List<Double> scoreValues =
+    // all sentiments built by comment content retrieved from youtubeResponse
+    List<Sentiment> documentList =
         youtubeResponse.getItems().stream()
             .map(UserComment::new)
-            .map(this::calcualateSentiAnalysisScore)
+            .map(
+                comment ->
+                    languageService
+                        .analyzeSentiment(
+                            Document.newBuilder()
+                                .setContent(comment.getCommentMsg())
+                                .setType(Document.Type.PLAIN_TEXT)
+                                .build())
+                        .getDocumentSentiment())
+            .collect(Collectors.toList());
+    // List of scores and magnitude for all comments
+    List<Double> scoreValues =
+        documentList.stream().map(this::calcualateSentiAnalysisScore).collect(Collectors.toList());
+    List<Double> magnitudeValues =
+        documentList.stream()
+            .map(this::calculateSentiAnalysisMagnitude)
             .collect(Collectors.toList());
```

Comment:
It would be better to pass a List of `Sentiment` objects to the `Statistics` class.

By splitting the scores and magnitudes, it becomes very difficult to associate the score of a specific comment with its magnitude.

To service front-end features like sorting comments of a specific sentiment bucket by their sentiment magnitude, this kind of association is necessary.